### PR TITLE
Support arguments binding.

### PR DESCRIPTION
**Pull request by take-cheeze**
Like `defer func() {}` pattern.

### CODE REVIEW

File: test/test_deferral.rb [hunk 1]
```diff
@@ -146,4 +146,17 @@ def close
     assert r1.closed?
     assert r2.closed?
   end
+
+  test 'pass arguments to defer' do
+    r1 = Resource.new
+    r2 = Resource.new
+    1.times do
+      r = r1
+      Deferral.defer(r){|r| r.close }
```
Comment (tagomoris) [inline test/test_deferral.rb]: Use non-`r` variable name as block parameter to show it's different from loop local variable.

File: lib/deferral.rb [hunk 2]
```diff
@@ -12,12 +12,12 @@ module Deferral
   # end
 
   module Mixin
-    def defer(&block)
+    def defer(*args, **kwds, &block)
```
Comment (tagomoris) [inline lib/deferral.rb]: `kwargs` is more general name in Ruby.

File: test/test_deferral.rb [hunk 3]
```diff
@@ -146,4 +146,17 @@ def close
...
+      r = r2
+      Deferral.defer(resource: r){|resource:| r.close }
```
Comment (tagomoris) [inline test/test_deferral.rb]: It seems that the variable name in deferred block should be `resource`, doesn't it?

**Comment by take-cheeze** [comment]
@tagomoris Thanks for the review. Should be fixed now.

**Review by tagomoris — APPROVED** [review]
LGTM!

**Pull request merged**